src/main/scala/com/databricks/spark/csv/package.scala

```diff
@@ -161,10 +169,16 @@ package object csv {
 
           override def next: String = {
             if (iter.nonEmpty) {
-              val row = csvFormat.format(iter.next().toSeq.map(_.asInstanceOf[AnyRef]): _*)
+              val values: Seq[AnyRef] = iter.next().toSeq.map(
```

Comment:
I changed to using an index they way you suggested. I ran some ad hoc tests using a 500k row dataset and saw the time go from about 5.5s to 4.9s (about 11% faster). There were 10 columns, and only 1 was a date column.